src/pybind/mgr/cephadm/module.py

```diff
@@ -1550,20 +1565,31 @@ def describe_service(self, service_type=None, service_name=None,
             # ugly sync path, FIXME someday perhaps?
             for host, hi in self.inventory.items():
                 self._refresh_host_daemons(host)
+        # <service_map>
         sm = {}  # type: Dict[str, orchestrator.ServiceDescription]
         for h, dm in self.cache.daemons.items():
             for name, dd in dm.items():
                 if service_type and service_type != dd.daemon_type:
                     continue
-                n = dd.service_name()
+                # <name> i.e. rgw.realm.zone
+                n: str = dd.service_name()
                 if service_name and service_name != n:
                     continue
+                service_store = PersistentStoreDict(self, f'service_spec/{dd.daemon_type}')
+                try:
+                    _ = service_store[dd.service_name(without_service_type=True)]
```

Comment:
don't we want the service name, to avoid ambiguity between say rgw.foo.bar and mds.foo.bar?

Reply:
It's included. I just went with clear and consistent hierarchy.

The `PersistentStoreDict` already loads `service_spec/$daemon_type` and then saves the individual name from `service_name()` under that namespace.

rgw:
- `mgr/cephadm/service_spec/rgw/realm1.zone1`
- `mgr/cephadm/service_spec/rgw/realm2.zone2`

mds:
- `mgr/cephadm/service_spec/mds/fs_name1`
- `mgr/cephadm/service_spec/mds/fs_name2`

{mgr, mon}:
- `mgr/cephadm/service_spec/{mgr, mon}/{mgr, mon or custom name}`

Reply:
Oh i see.  That works, but IMO it's simpler to have a flat namespace of "service name" to spec.  One instance of PersistentStoreDict (or whatever it is), and not without_service_type=True weirdness.